### PR TITLE
Stable sort release values by key

### PR DESCRIPTION
**Pull request by longquan0104**
Fixing this issue https://github.com/fluxcd/helm-controller/issues/675
Helm upgrade always happens without any changes
Clone from #676

### CODE REVIEW

File: api/v2beta1/helmrelease_types.go [hunk 1]
```diff
@@ -956,6 +958,31 @@ func HelmReleaseAttempted(hr HelmRelease, revision string, releaseRevision int,
 	return hr, changed
 }
 
+// HelmReleaseChanged returns if the HelmRelease has changed compared to the
+// provided values.
+func HelmReleaseChanged(hr HelmRelease, revision string, releaseRevision int, valuesChecksums ...string) bool {
```
[the remark this answers is not in the document]
Comment (hiddeco) [reply api/v2beta1/helmrelease_types.go:963]: This code shouldn't continue to exist for too long, as we will be moving away from any of the utility methods in the API package soon anyway. Which is why I simply restructured this to be done with it, while taking into account existing contracts due to SemVer compatibility and "being nice".

File: internal/util/util.go [hunk 2]
```diff
@@ -34,6 +37,77 @@ func ValuesChecksum(values chartutil.Values) string {
...
+	}
+	return fmt.Sprintf("%x", sha1.Sum(s))
+}
+
+// SortMapSlice recursively sorts the given goyaml.MapSlice by key.
+// This is used to ensure that the values checksum is the same regardless
+// of the order of the keys in the values file.
+func SortMapSlice(ms goyaml.MapSlice) {
+	sort.Slice(ms, func(i, j int) bool {
+		return fmt.Sprintf("%T.%v", ms[i].Key, ms[i].Key) < fmt.Sprintf("%T.%v", ms[j].Key, ms[j].Key)
```
[the remark this answers is not in the document]
Comment (longquan0104) [reply internal/util/util.go]: Sorry, that was a mistake from me. I was thinking that the key is an `interface{}` so when a change in key from one type to another type (e.g., from string to int) should be considered as a change and there should be an upgrade. But after double-checking, when converting to `MapSlice` the input only takes the `Key` as a string

[the remark this answers is not in the document]
Comment (longquan0104) [reply internal/util/util.go]: Sorry for pasting the issue link that makes spam. That shows how amateur I am. 
`Let's squash the commits` Do you mean squash all the commits which include commits of Hideco into one commit or just some nonsense commit of me?

[the remark this answers is not in the document]
Comment (hiddeco) [reply internal/util/util.go]: Do not worry about it, I did this for you.

**Review by hiddeco — APPROVED** [review]

**Pull request merged**